### ISSUE TITLE
Shared VPC support for subnet-level sharing

### ISSUE THREAD

**Issue by jeheyer** [terraform-google-modules/terraform-google-network#521]
### TL;DR

Orgs using Shared VPC may wish to selectively share specific subnet(s) with specific project(s).  The subnet object could have a `shared_projects` attribute which would be a list of project IDs to share to 

### Terraform Resources

```markdown
data.google_project - to retrieve project number from project ID

google_compute_subnetwork_iam_binding - to assign IAM permissions for the project service accounts to a specific subnetwork
```

### Detailed design

[https://github.com/aws2gcp/gcp-network-terraform/blob/main/vpc-network/shared_vpc.tf](https://github.com/aws2gcp/gcp-network-terraform/blob/main/vpc-network/shared_vpc.tf)

### Additional information

Also might throw in ability to share subnets to specific accounts/groups.  This is in the above code as a `shared_accounts` attribute to the subnet object, also a list of strings.

**Comment by imrannayer** [comment]
@jeheyer when you create a service project using [project factory](https://github.com/terraform-google-modules/terraform-google-project-factory) you can mention either share all VPCs in the host project or share a specific project.
